test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs

```diff
@@ -111,15 +111,23 @@ public void TracerSdkSetsActivityDataRequestBasedOnSamplingDecision()
                     .SetSampler(testSampler)
                     .Build();
 
-            testSampler.DesiredSamplingResult = new SamplingResult(SamplingDecision.RecordAndSampled);
+            testSampler.SamplingAction = (samplingParameters) =>
```

Comment:
side nit (and not directly from your change): the name `SamplingAction` is ambiguous, the older name was more precise. Anyway, the test sampler type is shared and the code here is clear on what it is doing. Ignore this note :)